### PR TITLE
Merge fields 

### PR DESCRIPTION
**Pull request by nvolk**
Merge fields within a record. Mainly used for merging otherwise identical fields, but with different 700$e values within into one (eg. for artikkelien syöttölomake, and old music fixes)

Code is transferred from melinda-marc-record-merge-reducers + added some sanity checks, and checks and balances required by the record-internal merge.

### CODE REVIEW

File: src/merge-fields/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+//import createDebugLogger from 'debug';
+import fs from 'fs';
+import path from 'path';
+import {mergeField} from './mergeField';
+import {MarcRecord} from '@natlibfi/marc-record';
+
+const description = 'Merge fields within record';
+
+// const multimediaRegexp = /multimedia/ui;
```
Comment (ammsalme) [inline src/merge-fields/index.js:18]: This is something old?

File: src/merge-fields/mergeField.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  if (!baseField.subfields) {
+    return;
+  }
+  // MELINDA-8978: prefer Asteri version
+  if (isAsteriField(sourceField) && !isAsteriField(baseField)) {
+    return 1;
+  }
+
+  function isAsteriField(field) {
+    if (field.subfields.some(sf => sf.code === '0' && sf.value.match(/^\((?:FI-ASTERI-N|FIN11)\)[0-9]{9}$/u))) {
```
Comment (ammsalme) [inline src/merge-fields/mergeField.js:130]: We'll probably want to handle also Asteri-W fields as preferable to non-Asteri fields. What to do with Asteri-N vs Asteri-W fields is more complicated.

Comment (nvolk) [reply src/merge-fields/mergeField.js]: Added.

File: src/merge-fields/index.js (hunk 1 above)
Comment (nvolk) [reply src/merge-fields/index.js]: Removed.

**Pull request merged**